### ISSUE TITLE
How do i get this working for an AAAA Record?

### ISSUE THREAD

**Issue by Tybil** [Domain-Connect/DomainConnectDDNS-Python#22]
I only have an IPv6 Address available, is there a way to alter this to set an AAAA-Record?

**Comment by pawel-kow** [comment]
Hi,
Thanks for a feature request. Seems that ipify.org suports also IPv6, so it can be potentially implemented.
The issue is that the current version of DNS template does not yet support it:
https://github.com/Domain-Connect/Templates/blob/master/domainconnect.org.dynamicdns.json

Let me issue a PR on the template first, then the feature can be added to the code.

**Comment by derwenno** [comment]
I have the same issue.

**Comment by pawel-kow** [comment]
PR with new template created.

**Comment by daviddavo** [comment]
The template is now merged, how is this going?

**Comment by pawel-kow** [comment]
A bit busy these days, but I can get the template deployed with IONOS today or tomorrow and open to review any community contribution :)

**Comment by robin-moser** [comment]
Hi @pawel-kow,

thanks for the PR. I tried to update the DDNS python client to work with the new template and did get it to work, but noticed a flaw in the template, which kinda defeats the purpose this ticket is for:

With my ISP, and probably lots of other users' providers, there is ONLY IPv6 available, I don't have a valid IPv4 address which I could connect to. But the new v2 template expects an IPv6 AND IPv4 address.

I think you should be able to select either v4, v6, or (default) both. 

The problem with both types is, that many clients will prefer requests to ipv4 if a valid A record is found. It doesn't matter if this address is actually valid. 

Would it be possible, to have a new template for only IPv6, so the templates called something like:
 - **domainconnect.org.dynamicdns-ipv6.json**  (new name, for only v6)
 - **domainconnect.org.dynamicdns.json**          (the old untouched template, only v4)
 - **domainconnect.org.dynamicdns-v2.json**     (the already existing new one, both v4, and v6)

If you agree on this, I can make the PR, too.

Regards, Robin

**Comment by pawel-kow** [comment]
Hi Robin, this would be supported by v2 template. Please note there are 2 group IDs defined so you can apply the template parts with ipv4 and ipv6 separately. 
Hope this helps.

**Comment by robin-moser** [comment]
Could you point me to the right direction?
I intercepted the traffic and became following response from the provider: (ip and domain redacted)

```
POST /v2/domainTemplates/providers/domainconnect.org/services/dynamicdns-v2/apply?domain=example.com&host=test&IPv6=0000%3A0000%3A0000%3A0000%3A0000%3A0000%3A0000%3A0000 HTTP/1.1
Host: api.domainconnect.1and1.com
```
```json
{ "HttpCode": 400 },
[
  {
    "code": "MISSING_REQUIRED_TEMPLATE_PARAMETERS",
    "message": "Parameters are missing for this template.",
    "parameters": {
      "requiredParameters": [
        "IPv4"
      ]
    }
  }
]
```

**Comment by robin-moser** [comment]
Never mind, i missed the GroupId parameter from the docs. I try the next days to get a working python client for the new template.

**Comment by pawel-kow** [comment]
Fixed with f1ca215.
Will go out with the next release
